web/packages/design/src/DataTable/useTable.ts

```diff
@@ -235,8 +253,9 @@ function sortAndFilter<T>(
     })
   );
   if (sort) {
-    if (sort.onSort) {
-      output.sort((a, b) => sort.onSort(a[sort.key], b[sort.key]));
+    const { onSort } = sort;
+    if (onSort) {
+      output.sort((a, b) => onSort(a, b));
```

Comment:
When it comes to sorting, by default the `sort.key` is `column.key`. But the column can choose a different field and pass it as `altSortKey`, which makes `sort.key` to be that key. Figuring out which _key_ is currently used is doable, but it would require writing a conditional type, which I think is not worth the effort. 
Instead, we can just pass the entire object in `onSort` callback so the callsites have more flexibility (so probably `altSortKey` will be no longer needed) plus the types are much simpler.

>is the key property needed at all?

`sort.key` is still used when you don't specify `onSort` function.